### PR TITLE
RHDEVDOCS-4280: Added a procedure to customize the user perspective

### PR DESCRIPTION
**Pull request by shipsing**
Purpose: To resolve this issue: https://issues.redhat.com/browse/RHDEVDOCS-4280

Aligned team: DevConsole

OCP version this PR applies to: 4.12 

Content for preview: https://51686--docspreview.netlify.app/openshift-enterprise/latest/web_console/customizing-the-web-console.html#odc-customizing-user-perspective_customizing-web-console

SME review completed: @debsmita1 @jerolimov 

QE review completed: @sanketpathak 

Peer review completed: @gabriel-rh

### CODE REVIEW

**Comment by shipsing** [comment]
@debsmita1 @jerolimov, Could you please help with SME review of this PR? Thanks

File: modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc [hunk 1]
```diff
@@ -0,0 +1,103 @@
+// Module included in the following assembly:
+//
+// * web_console/customizing-the-web-console.adoc
+
+:_content-type: PROCEDURE
+[id="odc-customizing-user-perspective_{context}"]
+= Customizing user perspective 
+
+The OCP web console provides two perspectives, namely *Administrator* and *Developer*. As a cluster administrator, you can show or hide a perspective to all users or a specific user role. Customizing the perspective ensures that users can view only the perspective that is applicable to their role and tasks, which avoids confusion. For example, hide the *Administrator* perspective from unprivileged users so that they cannot manage cluster resources, users, and projects or show the *Developer* perspective to users with the developer role so that they can create, deploy, and monitor applications.
```
Comment (jerolimov) [inline modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc:9]:
```suggestion
The OCP web console provides two perspectives by default, namely *Administrator* and *Developer*. More perspectives might be available depending on installed console plugins. As a cluster administrator, you can show or hide a perspective to all users or a specific user role. Customizing the perspective ensures that users can view only the perspective that is applicable to their role and tasks, which avoids confusion. For example, hide the *Administrator* perspective from unprivileged users so that they cannot manage cluster resources, users, and projects or show the *Developer* perspective to users with the developer role so that they can create, deploy, and monitor applications. 
```

File: modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+Each perspective includes the following mandatory parameters, which you can edit in the YAML view:
+
+* `id`: Defines the ID of the perspective to show or hide
+* `visibility`: Defines the state of the perspective along with access review checks, if needed 
+* `state`: Defines whether the perspective is enabled, disabled, or needs an access review check
+
+[NOTE]
+====
+By default, both *Administrator* and *Developer* perspectives are enabled. When you customize the user perspective, your changes are applicable to the entire cluster.
```
Comment (jerolimov) [inline modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc:21]:
```suggestion
By default, all perspectives are enabled. When you customize the user perspective, your changes are applicable to the entire cluster.
```

File: modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+----
+
+. Click *Save*.
+
+.Procedure to customize perspective using form view
+
+. In the *Administrator* perspective, navigate to *Administration* -> *Cluster Settings*.
+. Select the *Configuration* tab and click the *Console (operator.openshift.io)* resource.
+. Click *Actions* -> *Customize* on the right side of the page.
+. In the *General* settings, customize the *Administrator* and *Developer* perspectives by selecting one of the following options from the dropdown list:
```
Comment (jerolimov) [inline modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc:90]:
```suggestion
. In the *General* settings, customize the perspectives by selecting one of the following options from the dropdown list:
```

**Review by christoph-jerolimov — COMMENTED** [review]
Looks almost good to me as well. :+1:

But one note on "The OCP web console provides two perspectives"

That's just the default, OCP "provides by default two perspectives", but plugins can add extra perspectives. I don't know if you want "discuss" this in this context here.

If yes I added some comments below. If this should focus on the default with only an Admin and Dev perspective, this is already fine. :smile:

**Comment by shipsing** [comment]
> Looks almost good to me as well. +1
> 
> But one note on "The OCP web console provides two perspectives"
> 
> That's just the default, OCP "provides by default two perspectives", but plugins can add extra perspectives. I don't know if you want "discuss" this in this context here.
> 
> If yes I added some comments below. If this should focus on the default with only an Admin and Dev perspective, this is already fine. smile

I guess your suggestion brings in more clarity to the context here. So, i will definitely implement your suggestions and keeping the context generic would be helpful to avoid duplicate efforts in future.

**Review by christoph-jerolimov — APPROVED** [review]
Cool, thanks. Lgtm!

**Comment by shipsing** [comment]
@sanketpathak, Could you please help with QE review of this PR? Thanks

**Comment by shipsing** [comment]
@sanketpathak, Could you please help with QE review of this PR? Thanks

**Comment by sanketpathak** [comment]
LGTM

File: modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+.Procedure to customize perspective using YAML view
+
+. In the *Administrator* perspective, navigate to *Administration* -> *Cluster Settings*.
+. Select the *Configuration* tab and click the *Console (operator.openshift.io)* resource.
+. Click the *YAML* tab and make your customization: 
+.. To enable or disable a perspective, insert the snippet for *Add user perspectives* and edit the YAML code as needed:
++ 
+[source,yaml]
+----
+spec:
```
Comment (gabriel-rh) [inline modules/odc-customizing-user-perspective-using-CLI-and-UI.adoc:36]: The contrib guide say  to always include:

> Required fields
> Include the apiVersion and kind so that a user always knows the context of the YAML.
> 
> Include the full hierarchy to a deeply nested key.
> 
> For objects that are in the global scope, such as for config.openshift.io API group, always include the metadata.name for the object, which is usually cluster.

https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#yaml-formatting-for-kubernetes-and-openshift-api-objects

**Review by gabriel-rh — COMMENTED** [review]
- Some optional suggestions to simplify the wording in places
- Required fields for YAML. I realize most existing content does not follow these rules :-)
- I will update the review on multiple procedures once I get some feedback..

**Comment by shipsing** [comment]
@gabriel-rh I have incorporated your suggestions. please verify the changes and confirm.
Thanks

File: modules/odc-customization-of-user-perspectives.adoc [hunk 5]
```diff
@@ -0,0 +1,22 @@
+// Module included in the following assembly:
+//
+// * web_console/customizing-the-web-console.adoc
+
+:_content-type: CONCEPT
+[id="odc-customization-of-user-perspectives_{context}"]
+= Customization of user perspectives
```
Comment (gabriel-rh) [inline modules/odc-customization-of-user-perspectives.adoc:7]: The mod docs guide does say

"In such cases, ensure that the heading of the concept or reference remains a noun phrase and not a gerund. "

But in this case, maybe we should make an exception, especially seeing as we are using the CONCEPT as a workaround for not being having nested procedure modules. I think the title would look better as "Customizing user perspectives" in the toc, rather than as a noun phrase.

```
Adding a custom logo and product name
Creating custom links in the web console
Customizing console routes
  Customizing the console route
  Customizing the download route
Customizing the login page
Defining a template for an external log link
Creating custom notification banners
Customizing CLI downloads
Adding YAML examples to Kubernetes resources
Customization of user perspectives
  Customizing a perspective using YAML view
  Customizing a perspective using form view
```

**Review by gabriel-rh — COMMENTED** [review]
Great work! One small suggestion which requires going against the rules, but I think it is justified here

File: modules/odc-customization-of-user-perspectives.adoc (hunk 5 above)
Comment (shipsing) [reply modules/odc-customization-of-user-perspectives.adoc]: @gabriel-rh, Incorporated this suggestion. Looks completely justified.

**Review by gabriel-rh — APPROVED** [review]
/lgtm

Looks really good - well done!
